### PR TITLE
changes for sse s3.

### PR DESCRIPTION
**Pull request by mdw-at-linuxbox**
This is a set of commits to add testing support for sse-s3 functionality in https://github.com/ceph/ceph/pull/44494

There's currently one problem here, the sse-s3 tests should not be run when vault is not deployed.  There needs to be logic here or in ceph to make that happen, which doesn't exist yet.

### CODE REVIEW

File: s3tests_boto3/functional/test_s3.py [hunk 1]
```diff
@@ -12648,13 +12622,125 @@ def remove_header(**kwargs):
...
 @attr(method='put')
-@attr(operation='Deny put obj requests without encryption')
+@attr(operation='Deny put obj specifying both sse-c and sse-s3')
+@attr(assertion='success')
+@attr('encryption')
+def test_put_obj_enc_conflict_c_s3():
+    bucket_name = get_new_bucket()
+    client = get_v2_client()
+
+    boto3.set_stream_logger(name='botocore')
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:12632]: @mdw-at-linuxbox this and and rest of the iterations need to be deleted.

File: s3tests_boto3/functional/test_s3.py [hunk 2]
```diff
@@ -12685,102 +12771,249 @@ def test_bucket_policy_put_obj_enc():
...
-    response = client.put_object_tagging(Bucket=bucket_name, Key='invalidtag', Tagging=input_tagset)
-    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    p = Policy()
+    resource = _make_arn_resource("{}/{}".format(bucket_name, "*"))
+
+    s1 = Statement("s3:PutObject", resource, effect="Deny", condition=deny_incorrect_algo)
+    s2 = Statement("s3:PutObject", resource, effect="Deny", condition=deny_unencrypted_obj)
+    policy_document = p.add_statement(s1).add_statement(s2).to_json()
+
+    boto3.set_stream_logger(name='botocore')
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:12922]: @mdw-at-linuxbox So does this one

File: s3tests_boto3/functional/test_s3.py [hunk 3]
```diff
@@ -12648,13 +12622,125 @@ def remove_header(**kwargs):
...
+@attr(assertion='success')
+@attr('encryption')
+def test_put_obj_enc_conflict_c_kms():
+    kms_keyid = get_main_kms_keyid()
+    if kms_keyid is None:
+        kms_keyid = 'fool-me-once'
+    bucket_name = get_new_bucket()
+    client = get_v2_client()
+
+    boto3.set_stream_logger(name='botocore')
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:12662]: @mdw-at-linuxbox So does this one

File: s3tests_boto3/functional/test_s3.py [hunk 4]
```diff
@@ -12648,13 +12622,125 @@ def remove_header(**kwargs):
...
+@attr(assertion='success')
+@attr('encryption')
+def test_put_obj_enc_conflict_s3_kms():
+    kms_keyid = get_main_kms_keyid()
+    if kms_keyid is None:
+        kms_keyid = 'fool-me-once'
+    bucket_name = get_new_bucket()
+    client = get_v2_client()
+
+    boto3.set_stream_logger(name='botocore')
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:12693]: @mdw-at-linuxbox So does this one

File: s3tests_boto3/functional/test_s3.py [hunk 5]
```diff
@@ -12648,13 +12622,125 @@ def remove_header(**kwargs):
...
+@attr(assertion='success')
+@attr('encryption')
+def test_put_obj_enc_conflict_bad_enc_kms():
+    kms_keyid = get_main_kms_keyid()
+    if kms_keyid is None:
+        kms_keyid = 'fool-me-once'
+    bucket_name = get_new_bucket()
+    client = get_v2_client()
+
+    boto3.set_stream_logger(name='botocore')
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:12721]: @mdw-at-linuxbox So does this one

File: s3tests_boto3/functional/test_s3.py [hunk 6]
```diff
@@ -12685,102 +12771,249 @@ def test_bucket_policy_put_obj_enc():
...
     resource = _make_arn_resource("{}/{}".format(bucket_name, "*"))
 
-    s1 = Statement("s3:PutObject", resource, effect="Allow", condition=tag_conditional)
-    policy_document = p.add_statement(s1).to_json()
+    s1 = Statement("s3:PutObject", resource, effect="Deny", condition=deny_incorrect_algo)
+    s2 = Statement("s3:PutObject", resource, effect="Deny", condition=deny_unencrypted_obj)
+    policy_document = p.add_statement(s1).add_statement(s2).to_json()
 
-    client.put_bucket_policy(Bucket=bucket_name, Policy=policy_document)
+    boto3.set_stream_logger(name='botocore')
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:12826]: @mdw-at-linuxbox So does this one

File: s3tests_boto3/functional/test_s3.py [hunk 7]
```diff
@@ -12685,102 +12771,249 @@ def test_bucket_policy_put_obj_enc():
...
 
+    s1 = Statement("s3:PutObject", resource, effect="Deny", condition=deny_incorrect_algo)
+    s2 = Statement("s3:PutObject", resource, effect="Deny", condition=deny_unencrypted_obj)
+    policy_document = p.add_statement(s1).add_statement(s2).to_json()
 
-    client.put_bucket_policy(Bucket=bucket_name, Policy=policy_document)
-    tagset = []
-    tagset.append({'Key': 'security', 'Value': 'public'})
-    tagset.append({'Key': 'foo', 'Value': 'bar'})
+    boto3.set_stream_logger(name='botocore')
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:12875]: @mdw-at-linuxbox So does this one

**Comment by mdw-at-linuxbox** [comment]
I've just pushed one more commit for this - comments out the set_stream_logger calls above.

File: s3tests_boto3/functional/test_s3.py [hunk 8]
```diff
@@ -12931,7 +13164,7 @@ def test_object_lock_put_obj_lock_invalid_years():
 @attr(operation='Test put object lock with invalid mode')
 @attr(assertion='fails')
 @attr('object-lock')
-def test_object_lock_put_obj_lock_invalid_years():
+def test_object_lock_put_obj_lock_invalid_mode():
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_s3.py:13167]: @mdw-at-linuxbox why was this test name changed?

File: s3tests/functional/test_s3.py [hunk 9]
```diff
@@ -792,51 +792,6 @@ def test_bucket_policy_set_condition_operator_end_with_IfExists():
 def _make_arn_resource(path="*"):
     return "arn:aws:s3:::{}".format(path)
 
-@attr(resource='object')
-@attr(method='put')
-@attr(operation='Deny put obj requests without encryption')
-@attr(assertion='success')
-@attr('encryption')
-@attr('bucket-policy')
-def test_bucket_policy_put_obj_enc():
```
Comment (alimaredia) [inline s3tests/functional/test_s3.py:801]: What this test added into the boto3 version?

File: s3tests_boto3/functional/test_s3.py (hunk 8 above)
Comment (mdw-at-linuxbox) [reply s3tests_boto3/functional/test_s3.py:13167]: because the next test had the same name and this one was actually testing invalid years.

**Review by alimaredia — APPROVED** [review]

**Comment by mattbenjamin** [comment]
yay!

**Pull request merged**